tool/upload_coverage.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+# DO NOT enable -x as it will leak the `CODECOV_TOKEN` secret
+set -euo pipefail
+cd "$(dirname "$0")/.."
+
+if [ ! -v GITHUB_REPOSITORY ]; then
+    exit
+fi
```

Comment:
```suggestion
if [ ! -v GITHUB_REPOSITORY ]; then
    exit 1
fi
```
Always indicate that an error happened.